bin/spack

```diff
@@ -142,6 +146,13 @@ def _main(args, unknown_args):
         import spack.util.debug as debug
         debug.register_interrupt_handler()
 
+    if args.color and args.no_color:
+        tty.die("Do not set and cancel color")
+    if args.color:
+        tty.set_color(True)
+    elif args.no_color:
+        tty.set_color(False)
```

Comment:
The default is to colorize if `stdout.isatty()` is True, therefore the behaviour depends on whether you are piping. have redirected stdout, or not. The command line arguments override the default behaviour to either always or never colorize.

Good call on the mutually exclusive arguments, I have changed the code and pushed. I did try to copy the `ls` or `grep` style which allows `--color` or e.g. `--color always` but it always consumed the subcommand e.g. `invalid choice: 'find' choose from: alway, auto, never`. Let me know if this not appropriate.